source/docs/software/telemetry/writing-sendable-classes.rst

```diff
@@ -0,0 +1,71 @@
+Writing Your Own Sendable Classes
+=================================
+
+Since the ``Sendable`` interface only has one method, writing your own classes that implement ``Sendable`` (and thus automatically log values to and/or consume values from the dashboard) is extremely easy: just provide an implementation for the overrideable ``initSendable`` method, in which setters and getters for your class's fields are declaratively bound to key values (their display names on the dashboard).
+
+For example, here is the implementation of ``initSendable`` from WPILib's ``BangBangController``:
+
+.. tabs::
+
+    .. group-tab:: Java
+
+        .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpimath/src/main/java/edu/wpi/first/math/controller/BangBangController.java
+           :language: java
+           :lines: 150-158
+           :linenos:
+           :lineno-start: 150
+
+    .. group-tab:: C++
+
+        .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpimath/src/main/native/cpp/controller/BangBangController.cpp
+           :language: cpp
+           :lines: 58-72
+           :linenos:
+           :lineno-start: 58
+
+To enable the automatic updating of values by WPILib "in the background", ``Sendable`` data names are bound to getter and setter functions rather than specific data values.  If a field that you wish to log has no defined setters and getters, they can be defined inline with a lambda expression.
+
+The SendableBuilder Class
+-------------------------
+
+As seen above, the ``initSendable`` method takes a single parameter, ``builder``, of type ``SendableBuilder`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/util/sendable/SendableBuilder.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classwpi_1_1_sendablebuilder.html>`__).  This builder exposes methods that allow binding of getters and setters to dashboard names, as well as methods for safely ensuring that values consumed *from* the dashboard do not cause unsafe robot behavior.
+
+Databinding with addProperty Methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Like all WPILib dashboard code, ``Sendable`` fields are ultimately transmitted over :ref:`NetworkTables <docs/software/networktables/networktables-intro:What is NetworkTables>`, and thus the databinding methods provided by ``SendableBuilder`` match the supported NetworkTables data types:
+
+* ``boolean``: ``addBooleanProperty``
+* ``boolean[]``: ``addBooleanArrayProperty``
+* ``double``: ``addDoubleProperty``
+* ``double[]``: ``addDoubleArrayProperty``
+* ``string``:  ``addStringProperty``
+* ``string[]``: ``addStringArrayProperty``
+* ``byte[]``: ``addRawProperty``
+
+Ensuring Safety with setSafeState and setActuator
```

Comment:
Does this section belong here, or somewhere else, as setting hardware isn't really telemetry.

Reply:
I think it does belong here for now, as it's part of `Sendable` which largely exists for telemetry.  In the future we can split it off, maybe, but I don't think we have enough content yet for that to make sense.

Reply:
What about a new article in this section that could later be expanded?

Reply:
I think a stub would be a bit awkward; I'd rather leave it like this for now and reorganize it for 2023 when we have a clearer idea of what the landscape looks like.